docs(mm2): deciding when to use the heartbeat connector (#8459)
strimzi/strimzi-kafka-operator: documentation/modules/configuring/con-config-mirrormaker2-replication.adoc

== Deciding when to use the heartbeat connector

The heartbeat connector emits heartbeats to check connectivity between source and target Kafka clusters.
An internal `heartbeat` topic is replicated from the source cluster, which means that the heartbeat connector must be connected to the source cluster. 
The `heartbeat` topic is located on the target cluster, which allows it to do the following:

* Identify all source clusters it is mirroring data from
* Verify the liveness and latency of the mirroring process

This helps to make sure that the process is not stuck or has stopped for any reason. 
You enable the heartbeat connector by including its configuration in the MirrorMaker 2 deployment configuration.
However, using the heartbeat connector is not always necessary and can be left out of the MirrorMaker 2 configuration. 
For example, if your deployment has low network latency or a small number of topics, you might prefer to monitor the mirroring process using log messages or other monitoring tools. 

> If heartbeats are enabled, you can set the `emit.heartbeats.enabled` property of the heartbeat connector to `false` if you do not want to use them.

Reviewer: Shouldn't suggest to just removing the connector from the `KafkaMirrorMaker2` instead of setting some disable option? That would also correspond to the examples.

Author: Yes. I've updated

NOTE: If you have an application written in Java, you can use the `RemoteClusterUtils.java` utility to synchronize offsets through the application. The utility fetches remote offsets for a consumer group from the `checkpoints` topic. 

> == Deciding when to use the heartbeat connector

Reviewer: I agree with Mickael's comments -> but it seems good in general. However, as we removed the hearbeat connector from the examples -> do we also need to add an example of how to enable it to the docs?

Author: We show the heartbeat connector configuration in a later section, so I'm not sure we need it here.
I've intentionally kept the config out of this overview section so that it can be reused in rhel doc.
I did add this line though:

> You enable the heartbeat connector by including its configuration in the MirrorMaker 2 deployment configuration.

The heartbeat connector emits heartbeats to check connectivity between source and target Kafka clusters.
An internal heartbeat topic is replicated from the source cluster, which means that the heartbeat connector must be connected to the source cluster. 

> A target cluster uses the heartbeat topic to perform the following checks:

Reviewer 2: What about:
```suggestion
The heartbeat topic is located on the target cluster and allows to:
```

* Identify all source clusters it is mirroring data from
* Check that the Kafka Connect worker managing connectivity between clusters is running
* Verify that the source cluster is available

> You can use the heartbeat connector to monitor the liveness and latency of the mirroring process.

Reviewer 2: I wonder if we remove `Check that the Kafka Connect worker managing connectivity between clusters is running` and merge the last bullet point and following sentence into something like: `Verify the liveness and latency of the mirroring process.`

This helps to make sure that the process is not stuck or has stopped for any reason. 
However, using the heartbeat connector is not always necessary and can be left out of the MirrorMaker 2 configuration. 
For example, if your deployment has low network latency or a small number of topics, you might prefer to monitor the mirroring process using log messages or other monitoring tools. 

> Heartbeats are enabled by default, but you can set the `emit.heartbeats.enabled` property of the heartbeat connector to `false` if you do not want to use them.

Reviewer: Not sure what are we trying to say with this. All the example were modified, so the Heartbeat connector is not enabled by default anymore.
